### ISSUE TITLE
test failures due to type differences from readr 1.4.0

### ISSUE THREAD

**Issue by copernican** [metrumresearchgroup/rbabylon#287]
Several tests are failing as a consequence of changes in readr 1.4.0. In particular, tests involving the objects below fail because the `skip` element of the `spec` attribute is of type `numeric`, but `check_nonmem_table_output()` returns an `integer`.

```r
library(dplyr, warn.conflicts = FALSE)
library(purrr)

setwd("~/git/rbabylon/tests/testthat/")
ref_files <- c(
  "acop_ext_ref_floor0_200520.rds",
  "acop_ext_ref_floorNULL_200520.rds",
  "acop_grd_ref_floor0_200520.rds",
  "acop_grd_ref_floor10_200520.rds",
  "acop_grd_ref_floorNULL_200520.rds"
)

ref_files %>% 
  map(~ file.path("data", .)) %>% 
  map(readRDS) %>% 
  map(attr, "spec") %>% 
  map(`[[`, "skip") %>% 
  map_chr(typeof)
#> [1] "double" "double" "double" "double" "double"
```

The type difference is a consequence of a change to `read_table2()`, which is called by `check_nonmem_table_output()`, and previously returned a `numeric`.

```r
library(readr)
packageVersion("readr")
#> [1] '1.3.1'
x <- read_table2(
  "~/git/rbabylon/tests/testthat/model-examples/1/1.ext", 
  skip = 1
)
#> Parsed with column specification:
#> cols(
#>   ITERATION = col_double(),
#>   THETA1 = col_double(),
#>   THETA2 = col_double(),
#>   THETA3 = col_double(),
#>   THETA4 = col_double(),
#>   THETA5 = col_double(),
#>   `SIGMA(1,1)` = col_double(),
#>   `OMEGA(1,1)` = col_double(),
#>   `OMEGA(2,1)` = col_double(),
#>   `OMEGA(2,2)` = col_double(),
#>   OBJ = col_double()
#> )
str(attr(x, "spec")$skip)
#>  num 2
```

Note that this is unrelated to testthat 3.0.0.

```r
library(rbabylon)
library(testthat)

packageVersion("readr")
#> [1] '1.4.0'
packageVersion("testthat")
#> [1] '2.3.2'

setwd("~/git/rbabylon/tests/testthat/")
df <- check_nonmem_table_output("model-examples/1/1.ext", .x_var = "ITERATION")
ref_df <- readRDS("data/acop_ext_ref_floorNULL_200520.rds")
expect_identical(df, ref_df)
#> Error: `df` not identical to `ref_df`.
#> Objects equal but not identical
```

**Comment by dpastoor** [comment]
Great find and investigation thanks for getting to the bottom of it!

**Comment by seth127** [comment]
potentially fixed in https://github.com/metrumresearchgroup/rbabylon/pull/289 but we should verify

**Comment by seth127** [comment]
Closed by #289
